hpke/kemBase.go

```diff
@@ -0,0 +1,241 @@
+package hpke
+
+import (
+	"crypto"
+	_ "crypto/sha256" // linking sha256 packages.
+	_ "crypto/sha512" // linking sha512 packages.
+	"encoding"
+	"encoding/binary"
+	"io"
+
+	"github.com/cloudflare/circl/kem"
+	"golang.org/x/crypto/hkdf"
+)
+
+type dhKEM interface {
+	sizeDH() int
+	calcDH(dh []byte, sk kem.PrivateKey, pk kem.PublicKey) error
+	GenerateKey() (kem.PublicKey, kem.PrivateKey, error)
+	DeriveKey(seed []byte) (kem.PublicKey, kem.PrivateKey)
+	UnmarshalBinaryPrivateKey(data []byte) (kem.PrivateKey, error)
+	UnmarshalBinaryPublicKey(data []byte) (kem.PublicKey, error)
+}
+
+type kemBase struct {
+	id   KemID
+	name string
+	crypto.Hash
+	dh dhKEM
```

Comment:
`dh dhKEM` -> `dhKEM`? In other words, I think what we want to do is require `kemBase` to implement the `dhKEM` interface.

Reply:
we want kemBase implements what is common between xkem and shortkem.
the dh field in kemBase (the parent) allows to call child functions.